internal/statemachine/snap_states.go

```diff
@@ -42,3 +44,29 @@ func (stateMachine *StateMachine) prepareImage() error {
 
 	return nil
 }
+
+// Generate the manifest
+func (stateMachine *StateMachine) generateSnapManifest() error {
+	// We could use snapd's seed.Open() to generate the manifest here, but
+	// actually it doesn't make things much easier than doing it manually -
+	// like we did in the past. So let's just go with this.
+
+	// snaps.manifest
+	outputPath := filepath.Join(stateMachine.commonFlags.OutputDir, "snaps.manifest")
+	snapsDir := filepath.Join(stateMachine.tempDirs.rootfs, "system-data", "var", "lib", "snapd", "snaps")
```

Comment:
This causes errors with core20 builds, as this snapsDir doesn't exist. Perhaps you can check for

`if !stateMachine.isSeeded {}` before doing this?

Reply:
Or similar to the python code you could have the helper function exit without error if the `snapsDir` doesn't exist.

Reply:
Duh! Yes, completely missed that.